mkosi/__init__.py

```diff
@@ -4458,6 +4460,70 @@ def extract_unified_kernel(
     return f
 
 
+def extract_kernel_image_initrd(
+    args: MkosiArgs,
+    root: Path,
+    do_run_build_script: bool,
+    for_cache: bool,
+    mount: Callable[[], ContextManager[None]],
+) -> Tuple[Optional[BinaryIO], Optional[BinaryIO]]:
+    if do_run_build_script or for_cache or "linux" not in args.boot_protocols:
+        return None, None
+
+    prefix = "efi" if args.get_partition(PartitionIdentifier.esp) else "boot"
+
+    with mount():
+        kimgabs = None
+        initrd = None
+
+        for kver, kimg in gen_kernel_images(args, root):
+            kimgabs = root / kimg
+            initrd = root / prefix / args.machine_id / kver / "initrd"
+
+        if kimgabs is None:
+            die("No kernel image found, can't extract.")
+        assert initrd is not None
```

Comment:
```suggestion
        if kimgabs is None:
            die("No kernel image found, can't extract.")
        assert initrd is not None
```
That's a bit weird, no? For `kimgabs` we print a nice error and for `initrd` being `None` only an assertion error that would need to be figured out?